packages/e2e/test/artillery/StakePoolSearch.ts

```diff
@@ -0,0 +1,157 @@
+import * as envalid from 'envalid';
+import { ArtilleryContext, FunctionHook, WhileTrueHook } from './artillery';
+import { Cardano, Paginated, QueryStakePoolsArgs } from '@cardano-sdk/core';
+import { logger } from '@cardano-sdk/util-dev';
+import { stakePoolHttpProvider } from '@cardano-sdk/cardano-services-client';
+
+/**
+ * The context variables shared between all the hooks.
+ * Hooks must record here the state of the session of current artillery virtual user
+ */
+interface StakePoolSearchVars extends Paginated<Cardano.StakePool> {
+  /**
+   * The arguments used for the query; to repeat the same query, but next page, on next iteration
+   */
+  args: QueryStakePoolsArgs;
+
+  /**
+   * `performQuery()` stores here if it gets an error from the query to let `moreResults()` knows when exit on error
+   */
+  gotError?: boolean;
+
+  /**
+   * The result of the health check performed at the beginning of virtual user session
+   */
+  healthCheckResult: boolean;
+}
+
+const env = envalid.cleanEnv(process.env, { STAKE_POOL_PROVIDER_URL: envalid.str() });
+const provider = stakePoolHttpProvider({ baseUrl: env.STAKE_POOL_PROVIDER_URL, logger });
+
+const PAGE_SIZE = 20;
+
+/**
+ * The set of pools id found while getting results
+ */
+const poolIds: Cardano.PoolId[] = [];
+
+/**
+ * The artillery uid of the first artillery virtual user.
+ * Used to track time of first request to measure cache efficiency
+ */
+let firstUser = '';
+
+/**
+ * Gets the name of the stat to measure depending on the filter this user is using to perform the query
+ *
+ * @param ctx the artillery context
+ * @returns the name of the stat we are measuring
+ */
+const getStatName = (ctx: ArtilleryContext<StakePoolSearchVars>) => {
+  const { _uid, vars } = ctx;
+
+  if (_uid === firstUser) return 'query.First';
+
+  const { identifier, status } = vars.args.filters!;
+  const nameStatId = identifier ? 'id' : '__'; // ATM id is the only identifier used by the test
+  const statusStatName = status && status.length > 0 ? 'status' : '______';
+
+  return `query.${nameStatId}_${statusStatName}`;
+};
+
+const randomizeFilter = () => {
+  const filters: Exclude<QueryStakePoolsArgs['filters'], undefined> = {};
```

Comment:
There is a type utility for that:
```suggestion
  const filters: NonNullable<QueryStakePoolsArgs['filters']> = {};
```

Reply:
Thanks for the suggestion @mkazlauskas ! Applied with fc5eda72e